.github/workflows/ci.yml

```diff
@@ -17,15 +17,35 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
         check-latest: true
+
+    - name: Cache Python dependencies
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('pyproject.toml') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
```

Comment:
Yes - It falls back on the most recent cache hit if there's not an exact hit. This is actually what we want, because we always run `pip install`, regardless of a cache hit or miss!